Examples/Modules/qed/quantum_synchrotron/analysis.py

```diff
@@ -0,0 +1,298 @@
+    expected_photons_tolerance = 5.0*np.sqrt(expected_photons)
+    n_phot = ytdataset.particle_type_counts[phot_name]
+    assert( np.abs(n_phot-expected_photons) < expected_photons_tolerance)
+    print("  [OK] generated photons number is within expectations")
+    return n_phot
+
+def check_weights(part_data, phot_data):
+    assert(np.all(part_data["w"] == part_data["w"][0]))
+    assert(np.all(phot_data["w"]  == part_data["w"][0]))
+    print("  [OK] particles weights are the expected ones")
+
+def check_momenta(phot_data, p_phot, p0):
+    pdir = p0/np.linalg.norm(p0)
+    assert(small_diff(phot_data["px"]/p_phot, pdir[0]))
+    assert(small_diff(phot_data["py"]/p_phot, pdir[1]))
+    assert(small_diff(phot_data["pz"]/p_phot, pdir[2]))
+    print("  [OK] photons move along the initial particle direction")
+
+def check_opt_depths(part_data, phot_data):
+    data = (part_data, phot_data)
+    for dd in data:
+        loc, scale = st.expon.fit(dd["opt"])
+        assert( np.abs(loc - 0) < tol_red )
+        assert( np.abs(scale - 1) < tol_red )
+    print("  [OK] optical depth distributions are still exponential")
+
+def check_energy_distrib(gamma_phot, chi_part, gamma_part, n_phot, NN, idx):
+    gamma_phot_min = 1e-12*gamma_part
+    gamma_phot_max = gamma_part
+    h_log_gamma_phot, c_gamma_phot = np.histogram(gamma_phot, bins=np.logspace(np.log10(gamma_phot_min),np.log10(gamma_phot_max),NN+1))
+
+    cchi_phot_min = chi_part*(gamma_phot_min)/(gamma_part-1)
+    cchi_phot_max = chi_part*(gamma_phot_max)/(gamma_part-1)
+
+    #Rudimentary integration over npoints for each bin
+    npoints= 20
+    aux_chi = np.logspace(np.log10(cchi_phot_min),np.log10(cchi_phot_max), NN*npoints)
+    distrib = QS_d2N_dt_dchi(chi_part, gamma_part, aux_chi)*aux_chi
+    distrib = np.sum(distrib.reshape(-1, npoints),1)
+    distrib = n_phot*distrib/np.sum(distrib)
+
+    #c_gamma_phot = np.exp(0.5*(np.log(c_gamma_phot[1:])+np.log(c_gamma_phot[:-1])))
+
+    # Visual comparison of distributions
+    #plt.clf()
+
+    #fig, (ax1, ax2) = plt.subplots(1, 2)
+    #fig.suptitle("χ_particle = {:f}".format(chi_part))
+
+    #ax1.plot(c_gamma_phot, distrib,label="theory")
+    #ax1.loglog(c_gamma_phot, h_log_gamma_phot,label="QSR photons")
+    #ax1.set_xlim(1e-12*(gamma_part-1),gamma_part-1)
+    #ax1.set_ylim(1,1e5)
+
+    #ax2.plot(c_gamma_phot, distrib,label="theory")
+    #ax2.semilogy(c_gamma_phot, h_log_gamma_phot,label="QSR photons")
+    #ax2.set_ylim(1,1e5)
+    #ax2.set_xlim(1e-12*(gamma_part-1),gamma_part-1)
+
+    #ax1.set_xlabel("γ_photon")
+    #ax1.set_xlabel("N")
+
+    #ax2.set_xlabel("γ_photon")
+    #ax2.set_xlabel("N")
+
+    #plt.legend()
+    #plt.savefig("case_{:d}".format(idx+1))
+
+    discr = np.abs(h_log_gamma_phot-distrib)
+
+    max_discr = np.where(np.sqrt(distrib)>1, np.sqrt(distrib), 1)*5.0
+    # do not check the last 6 points: the lookup table is too coarse-grained for that
+    assert(np.all( np.abs(discr[:-6]) < max_discr[:-6]  ))
```

Comment:
```suggestion
    assert(np.all( discr[:-6] < max_discr[:-6]  ))
```
You probably don't need `np.abs` here (`discr` is already an absolute difference)

Reply:
You are right!